src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -144,52 +215,79 @@ sealed trait LazyModuleImpLike extends RawModule
+        * notice this place is not calling [[instantiate]], it just push the [[c.module]] to Builder.
+        * */
       val mod = Module(c.module)
+      /** ask each child to finish instantiate. */
       mod.finishInstantiate()
+      /** return dangles of each child. */
       mod.dangles
     }
+
+    /** ask each node in the [[LazyModule]] to call [[BaseNode.instantiate]].
+      * it will return a sequence of [[Dangle]] of these [[BaseNode]].
+      * */
     val nodeDangles = wrapper.nodes.reverse.flatMap(_.instantiate())
+    /** add all node and child dangle together. */
     val allDangles = nodeDangles ++ childDangles
+    /** internal nodes pairing, group [[allDangles]] which has same [[Dangle.source]]*/
     val pairing = SortedMap(allDangles.groupBy(_.source).toSeq:_*)
+    /** connect dangles can be paired. */
     val done = Set() ++ pairing.values.filter(_.size == 2).map { case Seq(a, b) =>
       require (a.flipped != b.flipped)
+      /** @todo quite strange. */
       if (a.flipped) { a.data <> b.data } else { b.data <> a.data }
       a.source
     }
+    /** find all not connected [[Dangle]] */
     val forward = allDangles.filter(d => !done(d.source))
+    /** generate [[AutoBundle]] io from [[forward]]. */
     val auto = IO(new AutoBundle(forward.map { d => (d.name, d.data, d.flipped) }:_*))
+    /** construct [[Dangle]] from [[auto]] make it can be accessed by father nodes. */
     val dangles = (forward zip auto.elements) map { case (d, (_, io)) =>
       if (d.flipped) { d.data <> io } else { io <> d.data }
       d.copy(data = io, name = wrapper.suggestedName + "_" + d.name)
     }
+    /** push all [[LazyModule.inModuleBody]] to [[chisel3.internal.Builder]]. */
     wrapper.inModuleBody.reverse.foreach { _() }
+    /** return [[IO]] and [[Dangle]] of this [[LazyModuleImp]]. */
     (auto, dangles)
   }
 
+  /** Ask each [[BaseNode]] in [[wrapper.nodes]] to call [[BaseNode.finishInstantiate]]
+    * notice: There are 2 different `finishInstantiate`:
+    *   [[LazyModuleImp.finishInstantiate]] and [[BaseNode.finishInstantiate]],
+    *   the former is a wrapper to latter.
+    * */
   protected[diplomacy] def finishInstantiate() {
     wrapper.nodes.reverse.foreach { _.finishInstantiate() }
   }
 }
 
 class LazyModuleImp(val wrapper: LazyModule) extends MultiIOModule with LazyModuleImpLike {
-  val (auto, dangles) = instantiate()
+  /** instantiate hardware of this `Module`. */
+  val (auto: AutoBundle, dangles: Seq[Dangle]) = instantiate()
 }
 
 class LazyRawModuleImp(val wrapper: LazyModule) extends RawModule with LazyModuleImpLike {
-  // These wires are the default clock+reset for all LazyModule children
-  // It is recommended to drive these even if you manually shove most of your children
-  // Otherwise, anonymous children (Monitors for example) will not be clocked
+  /** These wires are the default clock+reset for all LazyModule children.
+    * It will be used by clock/reset domain manger.
```

Comment:
```suggestion   
    * It is recommended to drive these even if you manually drive the [[clock]] and [[reset]] of all of the [[LazyRawModuleImp]] children.
    * Otherwise, anonymous children ([[Monitor]]s for example) will not have their [[clock]] and/or [[reset]] driven properly.
```